src/request-pipeline/file-request.ts

```diff
@@ -13,23 +14,36 @@ const TARGET_IS_NOT_FILE = 'The target of the operation is not a file';
 export default class FileRequest extends EventEmitter {
     url: string;
     path: string;
+    isAsarPath: boolean;
 
     constructor (url: string) {
         super();
 
-        this.url  = url;
-        this.path = FileRequest._getPath(url);
+        this.url        = url;
+        this.path       = FileRequest._getPath(url);
+        this.isAsarPath = isAsarPath(this.path);
 
         this._initEvents();
     }
 
     _initEvents () {
-        stat(this.path)
+        const pathToCheck = this.isAsarPath ? getArchivePath(this.path) : this.path;
+
+        stat(pathToCheck)
+            .catch((err: Error) => {
+                if (this.isAsarPath)
+                    throw new Error(`The asar archive target ("${getArchivePath(this.url)}") of the operation is not found`);
+
+                throw err;
+            })
             .then((stats: fs.Stats) => {
-                if (!stats.isFile())
-                    throw new Error(TARGET_IS_NOT_FILE);
+                if (!stats.isFile()) {
+                    throw new Error(this.isAsarPath
+                        ? `The asar archive target ("${getArchivePath(this.url)}") of the operation is not a file`
```

Comment:
I think we shouldn't throw an error if `isAsarPath === true` and `stats.isDirectory() === true`. It's a pretty ligitimate to use `.asar` in a name of a directory. Maybe we can just set `isAsarPath` to `false` in this case?